rclcpp/src/rclcpp/generic_publisher.cpp

```diff
@@ -31,4 +31,47 @@ void GenericPublisher::publish(const rclcpp::SerializedMessage & message)
   }
 }
 
+void GenericPublisher::publish_loaned_msg(const rclcpp::SerializedMessage & message)
+{
+  if (this->can_loan_messages()) {
+    auto loaned_message = borrow_loaned_message();
+    deserialize_message(message.get_rcl_serialized_message(), loaned_message);
+    publish_loaned_message(loaned_message);
+  } else {
+    throw std::runtime_error("failed to loan messages");
+  }
+}
+
+void * GenericPublisher::borrow_loaned_message()
+{
+  void * loaned_message = nullptr;
+  auto return_code = rcl_borrow_loaned_message(
```

Comment:
`can_loan_messages` should be issued before here to check if the feature is enabled on this publisher? because in that case, `failed to borrow loaned msg` is not really reasonable for user application?

Reply:
Yes. I will check return code to show correct reason.